File: 이것이 취업을 위한 코딩 테스트다/구현/문자열_재정렬/문자열_재정렬.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
  string s;
  vector<char> str;
  int result = 0 ;
  cin >> s;

  for (int i = 0; i < s.size(); i++) {
    if (isalpha(s[i])) {
      str.push_back(s[i]);
    } else {
      result += (s[i] - '0');
    }
  }

  sort(str.begin(), str.end());

  // string answer = "";

  // for (int i = 0; i < str.size(); i++) {
  //   answer += str[i];
  // }

  // cout << answer + to_string(result);

  // return 0;

  for (int i = 0; i < str.size(); i++) {
    cout << str[i];
  }

  if (result != 0) cout << result;
}

File: 이것이 취업을 위한 코딩 테스트다/구현/문자열_재정렬/문자열_재정렬.py
# s = input()
# result = 0
# string = []

# for i in s:
#   if i.isnumeric():
#     result += int(i)
#   else:
#     string.append(i)

# string.sort()
# print(''.join(string) + str(result))

data = input()
result = []
value = 0

for x in data:
  if x.isalpha():
    result.append(x)
  else:
    value += int(x)

result.sort()

if value != 0:
  result.append(str(value))

print(''.join(result))